mantidproject/mantid
# Add journal search to Reflectometry GUI

**Description of work.**

This PR adds a second method for finding metadata about runs in the ISIS Reflectometry GUI. It adds a method to the `QtCatalogSearcher` class to perform the search using `ISISJournalGetExperimentRuns` rather than `CatalogGetDataFiles`. This is required because the latter uses ICat, which is currently unreliable.

The journal search means we need to know the cycle name. This method is used if the cycle name is provided, and ICat is used if it is not. We may want to change this in future (and possibly improve functionality to provide a drop-down list of cycle names) but this is sufficient for a first cut at this functionality.

This PR also:
- Cleans up the `SearchModel` class so that metadata is parsed and errors set on construction (this logic has been moved out of `RunsPresenter`).
- Remove the Location field from the search results list because this isn't used.
- Clean up usage of `ITableWorkspace` - confine this to `QtCatalogSearcher`.

**Report to:** Max at ISIS

**To test:**

Must be tested by someone at ISIS. Ensure you have Search Data Archive enabled in the Manage User Directories dialog.

- Open the ISIS Reflectometry interface.
- Check that the new "Cycle" input is available on the Runs tab.
- Try performing a search with or without specifying the cycle, e.g. `INTER`, investigation `1920400`, cycle `19_4`. If the cycle is missing, you will be prompted for an ICat login -> although note that if you login here it will crash (fix coming in PR #28488). To get around this, run the `CatalogLogin` algorithm separately first to avoid the login via the GUI. Then the ICat search should work.
- When you click Search the list of runs for this experiment should be populated in the search results box. The results should be the same as the list you get from the `ISISJournalGetExperimentRuns` algorithm.
- Any runs without `th=` in the title (which specifies the angle) should be highlighted blue.
- Select some items in the results list and click `Transfer`. They should appear in the table on the right (you may need to expand the groups to see them), unless they are blue (invalid) rows in which case they will not be transferred.
- Click `Autoprocess` (and OK the warning if you get one). All valid rows should be transferred to the main table and will start processing. Ensure that the Cycle input is greyed out during processing. Note you must click Pause before to stop autoprocessing before you can exit the GUI.

Fixes #26897 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
